src/kartograf/filters/ring_changes.py

```diff
@@ -97,3 +97,59 @@ def filter_whole_rings_only(
         proposed_mapping = {v: k for k, v in filtered_mapping.items()}
 
     return proposed_mapping
+
+
+def filter_hybridization_rings(
+    molA: Chem.Mol, molB: Chem.Mol, mapping: dict[int, int]
+) -> dict[int, int]:
+    """Ensure that any mapped rings are either both aromatic or aliphatic"""
+
+    def get_atom_ring_hybridization_map(rdmol: Chem.Mol):
+        """
+        Helper function, getting molecule information about ring hybridizations.
+
+        Parameters
+        ----------
+        rdmol: Chem.Mol
+
+        Returns
+        -------
+        dict[int, list[bool]]:
+            returns a dict, that maps each atom to a list of ring
+            hybridizations it is in. (If True the ring is aromatic, if False
+            it is not entirely aromatic and therefore not necessarily
+            sterically restraint by a pi-orbital-system.)
+        """
+        riInf = rdmol.GetRingInfo()
+
+        # get_ring_hybridization
+        is_ring_aromatic = {}
+        for i, ring in enumerate(riInf.BondRings()):
+            is_ring_aromatic[i] = all(
+                rdmol.GetBondWithIdx(atomI).GetIsAromatic()
+                for atomI in ring)
+
+        # map atoms to rings:
+        atom_ring_map = defaultdict(list)
+        [atom_ring_map[a].append(is_ring_aromatic[ri]) for ri, r in
```

Comment:
hiding a loop inside a list comprehension is confusing, hiding two loops inside a list comprehension is worse.

I think this would be much clearer rewritten as two for loops